lib/src/common/cache/cache.dart

```diff
@@ -128,16 +140,19 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
   Stream<CacheContext<TIdentifier, TValue>> get didUpdate =>
       _didUpdateController.stream;
 
-  Iterable<TIdentifier> get keys => _cache.keys;
+  Iterable<TIdentifier> get keys =>
+      _cache.keys.where((TIdentifier key) => !_isReleased[key]);
 
-  Future<Iterable<TValue>> get values => Future.wait(_cache.values);
+  Future<Iterable<TValue>> get values => Future.wait(_cache.keys
+      .where((TIdentifier key) => !_isReleased[key])
```

Comment:
This logic is duplicated from the `keys` getter above, you could just reuse that here:

```dart
=> Future.wait(keys.map((TIdentifier key) => _cache[key]));
```

Unless you’d rather be explicit and not rely on the implementation of `keys` which could change.. I’m okay with it either way :)

Reply:
I like this because, in my mind, the implementation of `keys` and the implementation of `values` should always be consistent with one another.